### PR TITLE
UCP-3444: Media Feature: dark mode hocus; make whole card link

### PR DESCRIPTION
**Pull request by JBCSU**
# READY FOR REVIEW

# Summary

- Update the dark mode hocus styles to match that of the light mode. We don’t need a color difference in font because the text is not on a black background.
  - I also updated the dark mode book / audio icon to match light mode - same reason.
- Update so that the entire transparent card area is clickable.

# Review By

9/5/2024

# Criticality

1 - Minor

# Review Tasks

1. Visit https://news.stanford.edu/developreport/stories/jb-test-page
1. Confirm that the entire Media Feature component is a link
1. Confirm the link :hocus style is the same in dark mode as it is in light mode
    1. @iamrentman @broleomo Is it ok that the entire summary is underlined on `:hocus`? That looks weird to me.
1. Confirm the book icon  is the same in dark mode as it is in light mode

# Associated Issues and/or People

- [UCP-3444](https://stanfordits.atlassian.net/browse/UCP-3444)

[UCP-3444]: https://stanfordits.atlassian.net/browse/UCP-3444?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: components/media-feature/Component.jsx [hunk 1]
```diff
@@ -111,8 +112,8 @@ function MediaType({ type }) {
 function FeaturedHeading({ type }) {
   switch (type) {
     case "Podcast":
-      return <SidebarHeading icon="Featured audio" title="Featured audio" />;
```
Comment (iamrentman) [inline components/media-feature/Component.jsx:114]: This should be reverted. The text label next to the icon no longer prints as shown in the designs.

File: components/media-feature/Component.jsx [hunk 2]
```diff
@@ -2,9 +2,10 @@ import React from "react";
 
 // import specific templates for the component
 import { Container } from "../../packages/grids/Container";
-import { SidebarHeading } from "../../packages/headings/Heading";
```
Comment (iamrentman) [inline components/media-feature/Component.jsx:5]: I think we still need this. See comment below.

File: components/media-feature/Component.jsx [hunk 3]
```diff
@@ -39,49 +40,49 @@ export default function MediaFeature({
...
-              className={thumbMap.get(mediaType)}
-              // className="su-media-card-thumb su-size-full su-object-scale-down su-object-center"
-              // className="su-absolute su-object-cover su-rounded-[8px] su-z-[2] su-flex-1 su-size-full su-shadow-[0px_4px_7px_0px_rgba(0,0,0,0.15)]"
-            />
-          </div>
-
-          <div>
-            <div className="su-py-20 su-w-full md:su-pb-27 md:su-pt-0 *:dark:su-text-black">
-              <FeaturedHeading type={mediaType} />
+        <a href={linkUrl} className="su-media-feature-title-link su-group">
```
Comment (iamrentman) [inline components/media-feature/Component.jsx:43]: Place the `<a>` inside the `<h3>` below and add the `su-stretched-link` class – you can see this pattern in packages > card > verticalCard.jsx as reference. This keeps the link markup semantic while stretching the link across the container. (`su-relative` required on container div for this to work properly, see comment above)

File: components/media-feature/Component.jsx [hunk 4]
```diff
@@ -39,49 +40,49 @@ export default function MediaFeature({
   return (
     <Container width="full" paddingX={false}>
       <section className="su-py-45 su-px-20 su-flex su-justify-center su-relative md:su-py-72 md:su-px-50">
-        <div className="su-max-w-[1086px] su-flex su-flex-col su-items-center su-z-[2] su-relative su-p-38 before:su-content-[''] before:su-bg-foggy-light before:su-w-full before:su-h-full before:su-opacity-90 before:su-absolute before:su-z-[-1] before:su-top-0 before:su-left-0 md:su-flex-row md:su-gap-20 md:su-items-start lg:su-p-48 lg:su-gap-48">
```
Comment (iamrentman) [inline components/media-feature/Component.jsx:42]: For our `su-stretched-link` to work we need `su-relative` on this outer container of the "card".

Comment (iamrentman) [reply components/media-feature/Component.jsx:42]: Oh wait! `su-relative` is already there, nevermind.

However, for group hocus styles to work, add `su-group` class here.

File: components/media-feature/Component.jsx (hunk 3 above)
Comment (iamrentman) [reply components/media-feature/Component.jsx:43]: Remove `su-group` from this `<a>` element, add to container. See comment above.

File: components/media-feature/Component.jsx (hunk 1 above)
Comment (JBCSU) [reply components/media-feature/Component.jsx:114]: Done, but I changed <SidebarHeading>. See below.

File: components/media-feature/Component.jsx (hunk 2 above)
Comment (JBCSU) [reply components/media-feature/Component.jsx]: Done.

File: components/media-feature/Component.jsx (hunk 3 above)
Comment (JBCSU) [reply components/media-feature/Component.jsx]: Oh, `su-stretched-link` is so much mo' bettah. Thank you!

File: components/media-feature/Component.jsx (hunk 4 above)
Comment (JBCSU) [reply components/media-feature/Component.jsx]: I wasn't quite sure where you were saying that `su-group` should go, but I didn't change where it already was, and the hocus styles seem to work.

**Comment by JBCSU** [comment]
@iamrentman Thank you for your careful and helpful review! I think I responded to all your comments, with the possible exception of maybe not have the su-group class in the right place (see unresolved comment above). But hocus styles seem to work, so I think it's ok.

I did restore `<SidebarHeading>`, but I still really wanted the cardinal red (light mode) icon even in dark mode, since the icon renders on a light mode background regardless. So I tried a different way to render the light mode icon even in dark mode. That involved changing the `<SidebarHeading>` package in what you may feel is a heavy-handed way. Also, I think updating package means I need to update all components that use it? So it might not be worth it. I made the changes in a separate commit, so if you feel this is the wrong approach, I can back out that change and just make my peace with a green icon when in dark mode. :grinning: You get the final say.

Anyway, I deployed v2.0.3 of the component and updated [the test page in the develop env](https://news.stanford.edu/developreport/stories/jb-test-page), so it's ready for a second pass.

**Comment by iamrentman** [comment]
Regarding the icon changing color, I appreciate your prowess! I think we should let it change to the green gradient for consistency's sake, everything else does that.

**Comment by JBCSU** [comment]
I'll revert the icon color change. But I will point out that one of the two tasks in this ticket was to make the link hocus **not** consistent with what happens in dark mode. So it feels to me like, if the link stays in light mode but the icon switches to dark mode, then the component is not internally consistent, which hurts me a little. But not enough to fight for it. :wink: I'll revert.

But I learned stuff trying to make it work, so that's good.

**Review by iamrentman — APPROVED** [review]
Looks great! Thanks JB. Go for the merge 🚀

**Pull request merged**